libtenzir/src/node.cpp

```diff
@@ -330,6 +383,23 @@ node(node_actor::stateful_pointer<node_state> self, std::string /*name*/,
       }
     }
   });
+  self->set_exception_handler([=](std::exception_ptr& ptr) -> caf::error {
+    caf::error err;
+    try {
+      std::rethrow_exception(ptr);
+    } catch (const diagnostic& diag) {
+      err = diag.to_error();
+      TENZIR_ERROR("{}", err);
```

Comment:
If the node actor shuts down with an error we already print that in the `start_command` which started the node and waits for it to exit. This is going to cause diagnostics to be rendered twice.